en/codegen/vscode-plugin.md

```diff
@@ -0,0 +1,155 @@
+
+**angular.json**
+```json
+"@schematics/angular:component": {
+    "styleext": "scss"
+}
+```
+**.angular-cli.json**
+```json
+"defaults": {
+    "styleExt": "scss",
+}
+```
+
+This will tell the Angular CLI to generate SCSS instead of CSS when generating components which this extension will use.
+
+If you are working with an existing Angular project and it is already using the [Ignite UI for Angular](https://www.npmjs.com/package/igniteui-angular) package then you will need to update to at least version 6.1.0.  This extension may not update your project correctly if you have an older version.
+
+## How it works
+
+It all starts with a sketch.  Once a designer has completed their sketch they can upload it to the [Indigo.Design](https://next.indigo.design) website.  From there the designer can publish their work and create usability studies or open it up for commenting.  When the work is published Indigo.Design will provide a link to allow others to view your work.  That link can also be used by this extension.
+
+![](https://dl.infragistics.com/tools/extensions/codegen/step1.gif)
+
+With the share link the developer can see the various screens that make up the intended application.  From here the developer can select pieces of the screen that they want to generate code for.  Normally you would not generate the entire screen because the design will have duplicated elements on each screen.  We will refer to this as the application 'chrome'. Chrome refers to the parts of an application that do not change from screen to screen, such as navigation menus at the top of the page or even navigation drawers that would slide in from the side.
+
+![](https://dl.infragistics.com/tools/extensions/codegen/chrome.png)
+
+These elements should not be generated as they will be consistent elements across all screens of the application.  They only need to be written once to surround areas of the application that will be changing frequently through `<router-outlet>`.  Developers can use `<router-outlet>` in Angular to specify where content will be changing on the page.  This changing content is what you would generate code for.  Here's what that selection might look like:
+
+![](https://dl.infragistics.com/tools/extensions/codegen/step2.gif)
+
+Once things have been configured the way you like it is time to generate some code!  Click on the "Generate Code Assets" button and this will begin the process.
+
+Here is how the extension will update your project:
+
+* Install required NPM packages such as [Ignite UI for Angular](https://www.infragistics.com/products/ignite-ui-angular).
+* Create Angular components from selected sketch components.
+    * These will have the generated code.
+* Create modules around these components.
+* Create basic routes in the modules for each component to make them navigatable. (if configured, see Settings section)
```

Comment:
The content in the paragraph here is confusing.  What does it related to?  Why is it after the sentence?

Reply:
That section describes how the extension will update your project, not necessarily in that order.  Maybe it could be worded differently?

Reply:
Does it mean "Create basic routes in the modules (if configured, see Settings section) for each component to make them navigatable."?  I mean, which part has to be configured?  The component, or the modules?  Or the routes?